### PR TITLE
fix(core): remove hardcoded style from ngx for toolbar component

### PR DESCRIPTION
**Pull request by manu-kr**
#### Please provide a link to the associated issue.
Part of https://github.com/SAP/fundamental-ngx/issues/5025

#### Please provide a brief summary of this pull request.
Remove style from ngx lib as part of style cleanup for toolbar component.

## Breaking Changes:

- Added New Directory for toolbar over flow as part of style cleanup
`fd-toolbar-overflow-button`, `fd-toolbar-overflow-button-menu`

#### Toolbar overflow button 
`<button fd-toolbar-item fd-button label="Button" [compact]="true"></button>`

#### Toolbar overflow button
`<button fd-toolbar-item fd-button fd-toolbar-overflow-button label="Button" [compact]="true"></button>`

#### Toolbar overflow menu button
`<button fd-toolbar-item fd-button label="Button" [fdMenu]="true" [compact]="true"></button>`

#### Toolbar overflow menu button
`<button fd-toolbar-item fd-button fd-toolbar-overflow-button fd-toolbar-overflow-button-menu label="Button" [fdMenu]="true" [compact]="true"></button>`

### Before
Screenshot 2021-08-27 at 1 39 49 PM

### After
Screenshot 2021-08-27 at 1 40 51 PM

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [NA] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/0.32.0-Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/core/src/lib/toolbar/toolbar-overflow-button-menu.directive.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Directive } from '@angular/core';
+import { ToolbarItemDirective } from './toolbar-item.directive';
+
+@Directive({
+    // tslint:disable-next-line: directive-selector
+    selector: '[fd-toolbar-overflow-button-menu]',
```
Comment (Smiranin) [inline libs/core/src/lib/toolbar/toolbar-overflow-button-menu.directive.ts:6]: As I know we support two type of naming. camelCase and kebab-case. If you add camelCase option for the directive you no need to disable ts lint rule

**Review by N1XUS — APPROVED** [review]
Please fix @Smiranin comments

File: libs/core/src/lib/toolbar/toolbar-overflow-button-menu.directive.ts (hunk 1 above)
Comment (manu-kr) [reply libs/core/src/lib/toolbar/toolbar-overflow-button-menu.directive.ts]: Modified.

**Review by Betrozov — APPROVED** [review]

**Review by Smiranin — APPROVED** [review]

File: apps/docs/src/app/core/component-docs/toolbar/examples/toolbar-overflow-example.component.html [hunk 2]
```diff
@@ -43,18 +43,18 @@
         style="max-width: 200px;"
     />
 
-    <button fd-toolbar-item fd-button label="Button" [fdMenu]="true" [compact]="true"></button>
+    <button fd-toolbar-item fd-button fdToolbarOverflowButton fdToolbarOverflowButtonMenu label="Button" [fdMenu]="true" [compact]="true"></button>
```
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/toolbar/examples/toolbar-overflow-example.component.html:46]: Don't you find it weird to have 2 different naming conventions on the same element? `fd-button fdToolbarOverflowButton`????
Please use only one.

File: apps/docs/src/app/core/component-docs/toolbar/examples/toolbar-overflow-example.component.html [hunk 3]
```diff
@@ -43,18 +43,18 @@
...
-    <button fd-toolbar-item fd-button label="Button" [compact]="true"></button>
+    <button fd-toolbar-item fd-button fdToolbarOverflowButton label="Button" [compact]="true"></button>
+    <button fd-toolbar-item fd-button fdToolbarOverflowButton label="Button" [compact]="true"></button>
     <fd-toolbar-spacer fd-toolbar-item [fixed]="true" [width]="'200px'"></fd-toolbar-spacer>
-    <button fd-toolbar-item fd-button label="Button" [compact]="true"></button>
-    <button fd-toolbar-item fd-button label="Button" [compact]="true"></button>
-    <button fd-toolbar-item fd-button label="Button" [compact]="true"></button>
+    <button fd-toolbar-item fd-button fdToolbarOverflowButton label="Button" [compact]="true"></button>
+    <button fd-toolbar-item fd-button fdToolbarOverflowButton label="Button" [compact]="true"></button>
+    <button fd-toolbar-item fd-button fdToolbarOverflowButton label="Button" [compact]="true"></button>
```
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/toolbar/examples/toolbar-overflow-example.component.html:59]: See the comment above

File: apps/docs/src/app/core/component-docs/toolbar/toolbar-docs.component.html [hunk 4]
```diff
@@ -25,7 +25,7 @@
 <description>
    If there isn’t enough room in a Toolbar, the [shouldOverflow]="true" can be used to move buttons to an overflow area.
    A popover button will be created which will contain all the items in the overflow.
-   Make sure to include the <code class="code-snippet">fd-toolbar-item directive</code> in each Toolbar.
+   Make sure to include the <code class="code-snippet">fd-toolbar-item directive</code> in each Toolbar. <code class="code-snippet">fdToolbarOverflowButton</code> for button and <code class="code-snippet">fdToolbarOverflowButtonMenu</code> for menu button  in each Toolbar.
```
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/toolbar/toolbar-docs.component.html:28]: change with kebab-case example

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]
The Breaking changes examples serve no purpose to the user. They don't need screenshots of how the component looked before and now, they need code examples of what was changes so they can modify their code as fast/easily as possible.

File: apps/docs/src/app/core/component-docs/toolbar/examples/toolbar-overflow-example.component.html (hunk 2 above)
Comment (manu-kr) [reply apps/docs/src/app/core/component-docs/toolbar/examples/toolbar-overflow-example.component.html]: Updated example.

File: apps/docs/src/app/core/component-docs/toolbar/examples/toolbar-overflow-example.component.html (hunk 3 above)
Comment (manu-kr) [reply apps/docs/src/app/core/component-docs/toolbar/examples/toolbar-overflow-example.component.html]: Updated example.

File: apps/docs/src/app/core/component-docs/toolbar/toolbar-docs.component.html (hunk 4 above)
Comment (manu-kr) [reply apps/docs/src/app/core/component-docs/toolbar/toolbar-docs.component.html]: Updated.

**Comment by manu-kr** [comment]
> The Breaking changes examples serve no purpose to the user. They don't need screenshots of how the component looked before and now, they need code examples of what was changes so they can modify their code as fast/easily as possible.

Updated breaking changes.

**Comment by puru-hk** [comment]
Looks fine

**Review by puru-hk — APPROVED** [review]

**Review by Betrozov — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Review by mikerodonnell89 — APPROVED** [review]

**Comment by dimamarksman** [comment]
@manu-kr shouldn't we put **Breaking Changes** in the PR's description?

**Comment by manu-kr** [comment]
> @manu-kr shouldn't we put **Breaking Changes** in the PR's description?

> @dimamarksman As of now we not added **Breaking Changes** in description, Now i added it PR description as well.

**Pull request merged**